sqlitecluster/SQLiteNode.cpp

```diff
@@ -1930,8 +1930,15 @@ void SQLiteNode::_changeState(SQLiteNodeState newState) {
             _replicationThreadsShouldExit = true;
             uint64_t cancelAfter = _leaderCommitNotifier.getValue();
             SINFO("Replication threads should exit, canceling commits after current leader commit " << cancelAfter);
+            /*
             _localCommitNotifier.cancel(cancelAfter);
             _leaderCommitNotifier.cancel(cancelAfter);
+            */
+
+            // Hack. Some bug means that occasionally something doesn't get notified, and we wait forever in a loop.
+            // This needs to be fixed, but this probably un-breaks it in exchange for potentially losing in-flight transactions.
+            _localCommitNotifier.cancel(_db.getCommitCount());
+            _leaderCommitNotifier.cancel(_db.getCommitCount());
```

Comment:
This comment could benefit from some more detail in the comments - specifically, (1) from your PR description is great. Either delete the `cancelAfter` lines or match the comment style in the rest of the file.

NAB since this can be done as a follow up.